### ISSUE TITLE
Compiler errors

### ISSUE THREAD

**Issue by treedoug** [rar-ensemble/MATILDA.FT#5]
Dear developers,

When following the described build instructions, I get the following compiler error:

```
/usr/local/cuda-12/bin/nvcc -g -std=c++14 -rdc=true -O3 -I/usr/local/cuda-12/include -Wno-deprecated-gpu-targets -arch=sm_75 -c group.cu -o objects/group.cu.o
```

```
group.cu(38): error: namespace "thrust" has no member "device"
      thrust::sequence(thrust::device, d_index.begin(), d_index.end());
                               ^

group.cu(38): error: no instance of overloaded function "thrust::sequence" matches the argument list
            argument types are: (<error-type>, thrust::detail::normal_iterator<thrust::device_ptr<int>>, thrust::detail::normal_iterator<thrust::device_ptr<int>>)
      thrust::sequence(thrust::device, d_index.begin(), d_index.end());
      ^

group.cu(108): error: namespace "thrust" has no member "count"
          nsites = ns - thrust::count(d_all_id.begin(), d_all_id.end(), -1);
                                ^

group.cu(115): error: namespace "thrust" has no member "device"
          thrust::copy_if(thrust::device, d_all_id.begin(), d_all_id.end(), d_index.begin(), is_group_member());
                                  ^

group.cu(115): error: no instance of overloaded function "thrust::copy_if" matches the argument list
            argument types are: (<error-type>, thrust::detail::normal_iterator<thrust::device_ptr<int>>, thrust::detail::normal_iterator<thrust::device_ptr<int>>, thrust::detail::normal_iterator<thrust::device_ptr<int>>, Group::is_group_member)
          thrust::copy_if(thrust::device, d_all_id.begin(), d_all_id.end(), d_index.begin(), is_group_member());
          ^

5 errors detected in the compilation of "group.cu".
```

I get this error both with and without the "-arch=sm_75" command added to CFLAGS in the makefile.

System details:
* OS: Ubuntu 20.04.6 LTS
* CUDA 12.1
* GPU: Quadro RTX 4000
* gcc 9.4.0

**Comment by zuzanna-m-j** [comment]
Hi,

Thank you for the input.
I think it might be an issue with the line:
``CUDA_LOC = /usr/local/cuda-12`` (which I deduced from the error log)
It could be replaced with ``CUDA_LOC = /usr/local/cuda-12.1`` to match your CUDA version

I was able to compile the code on my machine.
My system details:

Operating System: Ubuntu 20.04.5 LTS

CUDA Version: 12.0   and 11.8 (tested on both by replacing  the path with ``CUDA_LOC = /usr/local/cuda-12``  and ``CUDA_LOC = /usr/local/cuda-11.8`` )

gcc version 9.4.0

GPU: Nvidia Quadro RTX 5000

This is my full makefile

```

# Copyright (c) 2023 University of Pennsylvania
# Part of MATILDA.FT, released under the GNU Public License version 2 (GPLv2).
#
####################################################
# Modify the line below to point to the CUDA library


CUDA_LOC = /usr/local/cuda-12

CC = ${CUDA_LOC}/bin/nvcc
# GIT_VERSION := $(shell git describe --abbrev=9 --dirty --always --tags)
# GIT_DIRECTORY := $(shell git rev-parse --show-toplevel)

EIGEN_LOC = ../include/eigen-3.4.0

CFLAGS = -std=c++14 -rdc=true -O3 -I${CUDA_LOC}/include -Wno-deprecated-gpu-targets
CPPFLAGS = -I${EIGEN_LOC} --expt-relaxed-constexpr

# CFLAGS = -std=c++11 -rdc=true -arch=sm_37 -dlto -O3 -I${CUDA_LOC}/include -lineinfo -g 

LIBS = -lm -lcudart -lcufft -L${CUDA_LOC}/lib

       
SRCS = array_utils.cu bonds.cu calc_properties.cu config_utils.cu cuda_random_posits.cu \
	device_array_utils.cu device_bonds.cu device_comm_utils.cu \
	device_grid_utils.cu device_utils.cu  field_component.cu forces.cu \
	initialize.cu integ_utils.cu io_utils.cu main.cu potential.cu \
	potential_fieldphases.cu potential_gaussian.cu pbc_utils.cu read_input.cu \
	potential_erf.cu reduce_utils.cu update_potentials.cu device_GJF_integrator.cu \
	potential_charges.cu potential_gaussian_erf.cu \
	Compute.cu angles.cu device_angles.cu random.cu grid_utils.cu \
	Extraforce.cu die.cu \
	Extraforce_dpd.cu Extraforce_langevin.cu Extraforce_lewis.cu Extraforce_midpush.cu Extraforce_wall.cu \
	group.cu \
	group_id.cu group_region.cu group_type.cu\
	integrator.cu \
	integrator_VV.cu integrator_EM.cu integrator_GJF.cu\
	nlist.cu \
	nlist_half_distance.cu nlist_bonding.cu \
	Compute_widom.cu Compute_avg_sk.cu Compute_sk.cu Compute_chempot.cu\
	tensor_potential.cu tensor_potential_MaierSaupe.cu device_ms_utils.cu \
	Box.cu FTS_Box.cu fts_species.cu fts_potential.cu \
	fts_potential_helfand.cu fts_potential_flory.cu fts_molecule.cu fts_molecule_linear.cu \
	fft_wrapper.cu run_fts_sim.cu fts_updateSchemes.cu \

SRCS_C = gsd.c
SRCS_CPP = tensor_potential_MaierSaupe_eigenvalue.cpp

OBJFOLDER := objects
OBJS_MAKE  	  = $(addprefix $(OBJFOLDER)/, ${OBJS})
OBJS_MAKE_C   = $(addprefix $(OBJFOLDER)/, ${OBJS_C})
OBJS_MAKE_CPP = $(addprefix $(OBJFOLDER)/, ${OBJS_CPP})

OBJS     = $(SRCS:.cu=.cu.o)
OBJS_C   = $(SRCS_C:.c=.c.o)
OBJS_CPP = $(SRCS_CPP:.cpp=.cpp.o)

$(OBJFOLDER)/%.cu.o: %.cu git-version.h
	@mkdir -p $(dir $@)
	${CC} -g ${CFLAGS} -c $< -o $@

$(OBJFOLDER)/%.cpp.o: %.cpp git-version.h
	@mkdir -p $(dir $@)
	${CC} -g ${CFLAGS} ${CPPFLAGS} -c $< -o $@

$(OBJFOLDER)/%.c.o: %.c git-version.h
	@mkdir -p $(dir $@)
	${CC} -g ${CFLAGS} -c $< -o $@
	 
../matilda.ft: $(OBJS_MAKE) $(OBJS_MAKE_CPP) $(OBJS_MAKE_C)
	$(CC) $(CFLAGS) -o $@ $(OBJS_MAKE) $(OBJS_MAKE_CPP) $(OBJS_MAKE_C) $(LIBS) -g

git-version.h: 
	sed "s/@GIT_VERSION@/$(GIT_VERSION)/"  git-version.h.in > git-version.h

clean:
	rm -fr *.o objects
	rm -f matilda.ft ../matilda.ft
	rm -f git-version.h
	rm -f *~

```

Please, let us know if there are more issues with compiling the code

**Comment by zuzanna-m-j** [comment]
Thank you for noticing that issue.
This has not been an issue with the versions of CUDA tested (11.8) but when I tried compiling with CUDA 12.0 I was able to reproduce the error. It does seem that the Thrust header files that have been automatically included previously are not included in the newer version. The issue has been resolved by including these files manually in the source files (compiled with CUDA 12.1) . We will be pushing the changes to the main branch.
